core/kernel/console.c

```diff
@@ -24,7 +25,10 @@ __weak void plat_console_init(void)
 
 void console_init(void)
 {
-	plat_console_init();
+	if (IS_ENABLED(CFG_SEMIHOSTING_CONSOLE))
+		semihosting_console_init(CFG_SEMIHOSTING_CONSOLE_FILE);
```

Comment:
This may cause link errors if compiling without optimizations and `CFG_SEMIHOSTING_CONSOLE` isn't enabled.

Reply:
`CFG_SEMIHOSTING_CONSOLE_FILE` has a default value so it should work.
Maybe _mk/config.mk_ should check the config switch is not empty.

Reply:
I'm concerned with the lack of a `semihosting_console_init()` in that case.

Reply:
Hi @jenswi-linaro 
Do you mean build optee_os with `CFG_CC_OPT_LEVEL=0` ?
I can see some items in CI build optee_os with `CFG_CC_OPT_LEVEL=0`
https://github.com/OP-TEE/optee_os/blob/7f124eb8587bd4fc11192ef970f4f34a26f19c21/.github/workflows/ci.yml#L114

Reply:
That's interesting. If it works then I don't object.